src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -253,8 +253,13 @@ public async Task<ChainedHeaderBlock> BlockMinedAsync(Block block)
                         ConnectBlocksResult fullValidationResult = await this.FullyValidateLockedAsync(partialValidationResult.ChainedHeaderBlock).ConfigureAwait(false);
                         if (!fullValidationResult.Succeeded)
                         {
-                            this.logger.LogError("Miner produced an invalid block, full validation failed: {0}", fullValidationResult.Error.Message);
-                            this.logger.LogTrace("(-)[FULL_VALIDATION_FALIED]");
+                            lock (this.peerLock)
+                            {
+                                this.chainedHeaderTree.PartialOrFullValidationFailed(chainedHeader);
+                            }
+
+                            this.logger.LogTrace("Miner produced an invalid block, full validation failed: {0}", fullValidationResult.Error.Message);
+                            this.logger.LogTrace("(-)[FULL_VALIDATION_FAILED]:null");
```

Comment:
not `:null` here, exception is going out, no return value

Reply:
Fixed